src/messaging/ExchangeTransport.cpp

```diff
@@ -0,0 +1,133 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *      This file provides implementation of ExchangeTransport class.
+ */
+
+#ifndef __STDC_FORMAT_MACROS
+#define __STDC_FORMAT_MACROS
+#endif
+
+#ifndef __STDC_LIMIT_MACROS
+#define __STDC_LIMIT_MACROS
+#endif
+
+#include <inttypes.h>
+
+#include <messaging/ExchangeTransport.h>
+#include <messaging/ReliableMessageContext.h>
+#include <messaging/ReliableMessageMgr.h>
+#include <protocols/secure_channel/Constants.h>
+#include <support/CodeUtils.h>
+
+namespace chip {
+namespace Messaging {
+
+CHIP_ERROR ExchangeTransport::SendMessage(SecureSessionHandle session, ExchangeTransport::ExchangeInfo & exchangeInfo,
```

Comment:
It seems ExchangeTransport::ExchangeInfo & exchangeInfo, bool isReliableTransmission,                             Protocols::Id protocol, uint8_t type are all used to construct PayloadHeader, why not just pass PayloadHeader?

Reply:
I considered using the `PayloadHeader` as the type. The issue is that not all fields of this structure are available/filled when these functions are called. At that point it's more of an understanding between the caller and callee that certain fields of the provided structure should not be accessed. This understanding cannot be expressed or enforced in the code.

Reply:
Updated the receive handler to take payload header as the parameter.
Updated SendMessage to take only required parameters, instead of the blanket structure. The payload header is constructed inside send message, so it made sense to keep the logic in there.

Reply:
The payload header should be constructed by the exchange user(protocols), we introduce sendFlags to hide the internal structure of payload header from the protocol perspective.